packages/core/src/domain/sessionManagement.ts

```diff
@@ -74,15 +75,54 @@ export function startSessionManagement<TrackingType extends string>(
     sessionCookie.clearCache()
     const session = retrieveActiveSession(sessionCookie)
     persistSession(session, sessionCookie)
+    if (session.id === inMemorySession.id && session[productKey] !== inMemorySession[productKey]) {
+      addMonitoringMessage('session type changed - es', {
+        debug: {
+          product: productKey,
+          inMemorySession,
+          retrievedSession: session,
+          newTrackingType: session[productKey],
+          _dd_s: getCookie(SESSION_COOKIE_NAME),
+        },
+      })
+      inMemorySession = { ...session }
+    }
   }
 
   expandOrRenewSession()
   trackActivity(expandOrRenewSession)
   trackVisibility(expandSession)
+  checkCookieConsistency()
+
+  function checkCookieConsistency() {
+    const alternateSessionCookie = cacheCookieAccess(SESSION_COOKIE_NAME, options)
+    const cookieConsistencyCheckInterval = setInterval(() => {
+      const sessionCookieCheck = retrieveActiveSession(alternateSessionCookie)
+      alternateSessionCookie.clearCache()
+      if (
+        inMemorySession.id === sessionCookieCheck.id &&
+        inMemorySession[productKey] !== undefined &&
+        inMemorySession[productKey] !== sessionCookieCheck[productKey]
+      ) {
+        addMonitoringMessage('session type changed - ccc', {
+          debug: {
+            product: productKey,
+            inMemorySession,
+            retrievedSession: sessionCookieCheck,
+            newTrackingType: sessionCookieCheck[productKey],
+            _dd_s: getCookie(SESSION_COOKIE_NAME),
+          },
+        })
+        inMemorySession = { ...sessionCookieCheck }
+      }
```

Comment:
NIT: This could be factorized. Since it is temporary code it may be fine though.

Reply:
yep, the goal is to remove this code ASAP, so IMO no need to put too much though into its maintainability